### PR TITLE
JDK-8295914: Add a header to generated HTML files in specs

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review some updates to the parts of the build system related to building the documentation.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295914](https://bugs.openjdk.org/browse/JDK-8295914): Add a header to generated HTML files in specs

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to [c5e567fd](https://git.openjdk.org/jdk/pull/10891/files/c5e567fd0b205778a369d086524f0aaef1adb280)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10891/head:pull/10891` \
`$ git checkout pull/10891`

Update a local copy of the PR: \
`$ git checkout pull/10891` \
`$ git pull https://git.openjdk.org/jdk pull/10891/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10891`

View PR using the GUI difftool: \
`$ git pr show -t 10891`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10891.diff">https://git.openjdk.org/jdk/pull/10891.diff</a>

</details>

### CODE REVIEW

**Review by erikj79 — COMMENTED** [review]
Build changes look fine in general.

**Review by irisclark — APPROVED** [review]
Your changes look good!  Can't wait to see this in a promotion.

File: make/Docs.gmk [hunk 1]
```diff
@@ -593,6 +593,9 @@ ifeq ($(ENABLE_PANDOC), true)
   # html, if we have pandoc (otherwise we'll just skip this).
 
   GLOBAL_SPECS_DEFAULT_CSS_FILE := $(DOCS_OUTPUTDIR)/resources/jdk-default.css
+  # Unset the following to suppress the link to the tool guides
+  NAV_LINK_GUIDES = --nav-link-guides
+  HEADER_RIGHT_SIDE_INFO = '<strong>$(subst &amp;,&,$(JDK_SHORT_NAME))$(DRAFT_MARKER_STR)</strong>'
```
[the remark this answers is not in the document]
Comment (jonathan-gibbons) [reply make/Docs.gmk:598]: Thanks for the tip and style guideline

**Review by erikj79 — APPROVED** [review]

**Pull request closed without merging**